rcl_lifecycle/src/default_state_machine.c

```diff
@@ -691,6 +691,7 @@ rcl_lifecycle_init_default_state_machine(
   return ret;
 
 fail:
+  // if rcl_lifecycle_transition_map_fini() fails, it will clobber the error string here, twice
```

Comment:
is that meant as a TODO here? If `fini()` fails and sets its own error message, I think we can either reset it here or concatenate. I think overwriting the error message actually yields a warning.

Reply:
No it was an issue I had found, and didn't have a great way of resolving it. I didn't know concatenation would work, I'll add that in. 

It will indeed print a warning if the error message is set and then it proceeds to set another error message. However, I didn't want to lose the trace of errors messages as they fall from one failure to the next (so it would have been impossible to see the original failure case).